sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/AbstractKafkaPropertiesBeanPostProcessor.java

```diff
@@ -0,0 +1,195 @@
+     * @return a Map containing all Kafka admin properties
+     */
+    protected abstract Map<String, Object> getMergedAdminProperties(T properties);
+
+    /**
+     * Get the raw {@link Map} object from the Kafka Spring properties that stores admin-specific properties.
+     * @param properties the Kafka Spring properties
+     * @return the map from Kafka properties storing admin-specific properties
+     */
+    protected abstract Map<String, String> getRawAdminProperties(T properties);
+
+    protected abstract boolean needsPostProcess(Object bean);
+
+    protected abstract Logger getLogger();
+
+    /**
+     * Process Kafka Spring properties for any customized operations.
+     * @param properties the Kafka Spring properties
+     */
+    protected void customize(T properties) {
+    }
+
+
+    protected void clearAzureProperties(Map<String, String> properties) {
+        AzureKafkaPropertiesUtils.AzureKafkaPasswordlessPropertiesMapping.getPropertyKeys()
+            .forEach(properties::remove);
+    }
+
+    /**
+     * This method executes two operations:
+     * <p>
+     * 1. When this configuration meets Azure Kafka passwordless startup requirements, convert all Azure properties
+     * in Kafka to {@link Jaas}, and configure the JAAS configuration back to Kafka.
+     * </p>
+     * <p>
+     * 2. Clear any Azure properties in Kafka properties.
+     * </p>
+     * @param mergedProperties the merged Kafka properties which can contain Azure properties to resolve JAAS from
+     * @param rawPropertiesMap the raw Kafka properties Map to configure JAAS to and remove Azure Properties from
+     */
+    private void replaceAzurePropertiesWithJaas(Map<String, Object> mergedProperties, Map<String, String> rawPropertiesMap) {
+        resolveJaasForAzure(mergedProperties)
+            .ifPresent(jaas -> {
+                configJaasToKafkaRawProperties(jaas, rawPropertiesMap);
+                logConfigureOAuthProperties();
+                configureKafkaUserAgent();
+            });
+        clearAzureProperties(rawPropertiesMap);
+    }
+
+    private Optional<Jaas> resolveJaasForAzure(Map<String, Object> mergedProperties) {
+        if (needConfigureSaslOAuth(mergedProperties)) {
+            Jaas jaas = JaasResolver.resolve(getSaslJaasFromKafkaOrDefault(mergedProperties));
+            setAzurePropertiesToJaasOptionsIfAbsent(azureGlobalProperties, jaas);
+            setKafkaPropertiesToJaasOptions(mergedProperties, jaas);
+            jaas.getOptions().put(AZURE_CONFIGURED_JAAS_OPTIONS_KEY, AZURE_CONFIGURED_JAAS_OPTIONS_VALUE);
+            return Optional.of(jaas);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private String getSaslJaasFromKafkaOrDefault(Map<String, Object> mergedProperties) {
+        String jaas = (String) mergedProperties.get(SASL_JAAS_CONFIG);
+        return StringUtils.hasText(jaas) ? jaas : DEFAULT_SASL_JAAS_CONFIG_OAUTH;
```

Comment:
Why can't this property be defined in this class?

Reply:
Here is because I have put most of SASL-config related constants in `AzureKafkaConfigurationUtils` class, so I defined this one there as well.